packages/discord.js/typings/index.d.ts

```diff
@@ -559,6 +560,11 @@ export class SelectMenuOptionBuilder extends BuildersSelectMenuOption {
   public setEmoji(emoji: ComponentEmojiResolvable): this;
 }
 
+export class ModalBuilder extends BuildersModal {
+  public constructor(data?: ModalData | APIModalComponent);
```

Comment:
The `ModalData` interface needs to be updated to allow both json components and class components